packages/react-components/react-rating-preview/src/components/Rating/useRating.tsx

```diff
@@ -24,15 +24,23 @@ export const useRating_unstable = (props: RatingProps, ref: React.Ref<HTMLDivEle
   const generatedName = useId('rating-');
   const {
     color = 'neutral',
-    iconFilled = <StarFilled />,
-    iconOutline = <StarRegular />,
+    iconFilled: iconFilledProp,
+    iconOutline: iconOutlineProp,
     max = 5,
     name = generatedName,
     onChange,
     step = 1,
     size = 'extra-large',
   } = props;
 
+  const iconFilled = React.useMemo(() => {
+    return iconFilledProp ?? <StarFilled />;
+  }, [iconFilledProp]);
+
+  const iconOutline = React.useMemo(() => {
+    return iconOutlineProp ?? <StarRegular />;
+  }, [iconOutlineProp]);
```

Comment:
Could these be module-level constants instead of being memoized for each instance?

(outside of this function): 
```jsx
const defaultIconFilled = <StarFilled />;
const defaultIconOutline = <StarRegular />;
```

Reply:
I think so. Are you suggesting we change this to 

`const iconOutline = iconOutlineProp ?? defaultIconOutline;`

as well? Or keep the memo?

Reply:
Should probably keep the memo still right?

Reply:
The reason to use the module-level constant is so we don't need `useMemo`. The default icons don't have any dependencies on props or other state, so they don't need to be created and memoized per-instance.

Lines 27-28 would be:
```diff
-    iconFilled = <StarFilled />,
-    iconOutline = <StarRegular />,
+    iconFilled = defaultIconFilled,
+    iconOutline = defaultIconOutline,
```